waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -50,52 +51,71 @@ validateExactlyOneAppExists spec =
 
 validateAppAuthIsSetIfAnyPageRequiresAuth :: AppSpec -> [ValidationError]
 validateAppAuthIsSetIfAnyPageRequiresAuth spec =
-  if anyPageRequiresAuth && not (isAuthEnabled spec)
-    then
-      [ GenericValidationError
-          "Expected app.auth to be defined since there are Pages with authRequired set to true."
-      ]
-    else []
+  [ GenericValidationError
+      "Expected app.auth to be defined since there are Pages with authRequired set to true."
+    | anyPageRequiresAuth && not (isAuthEnabled spec)
+  ]
   where
     anyPageRequiresAuth = any ((== Just True) . Page.authRequired) (snd <$> AS.getPages spec)
 
 validateDbIsPostgresIfPgBossUsed :: AppSpec -> [ValidationError]
 validateDbIsPostgresIfPgBossUsed spec =
-  if isPgBossJobExecutorUsed spec && not (isPostgresUsed spec)
-    then
-      [ GenericValidationError
-          "Expected app.db.system to be PostgreSQL since there are jobs with executor set to PgBoss."
-      ]
-    else []
+  [ GenericValidationError
+      "Expected app.db.system to be PostgreSQL since there are jobs with executor set to PgBoss."
+    | isPgBossJobExecutorUsed spec && not (isPostgresUsed spec)
+  ]
 
 validateAuthUserEntityHasCorrectFieldsIfUsernameAndPasswordAuthIsUsed :: AppSpec -> [ValidationError]
 validateAuthUserEntityHasCorrectFieldsIfUsernameAndPasswordAuthIsUsed spec = case App.auth (snd $ getApp spec) of
   Nothing -> []
   Just auth ->
-    if Auth.UsernameAndPassword `notElem` Auth.methods auth
+    if not $ Auth.isUsernameAndPasswordAuthEnabled auth
       then []
       else
         let userEntity = snd $ AS.resolveRef spec (Auth.userEntity auth)
             userEntityFields = Entity.getFields userEntity
-            maybeUsernameField = find ((== "username") . Entity.Field.fieldName) userEntityFields
-            maybePasswordField = find ((== "password") . Entity.Field.fieldName) userEntityFields
-         in concat
```

Comment:
Tried to DRY this up, especially since I needed many checks below.

Reply:
Looks good to me @shayneczyzewski !!!